src/odemis/dataio/tiff.py

```diff
@@ -1736,6 +1742,141 @@ def open_data(filename):
     return AcquisitionDataTIFF(filename)
 
 
+class DataArrayShadowTIFF(DataArrayShadow):
+    """
+    This class contains information about a DataArray.
+    It has all the useful attributes of a DataArray, but not the actual data.
+    """
+    def __init__(self, tiff_info, *args,**kwargs):
+        """
+        Constructor
+        tiff_info (dictionary): Information about the source tiff file and directory from which
+            the image should be read. It has 2 values:
+            'tiff_file' (handle): Handle of the tiff file
+            'dir_index' (int): Index of the directory
+        """
+        DataArrayShadow.__init__(self, *args, **kwargs)
+
+        if isinstance(tiff_info, list):
+            tiff_handle = tiff_info[0]['handle']
+        else:
+            tiff_handle = tiff_info['handle']
+
+        num_tcols = tiff_handle.GetField(T.TIFFTAG_TILEWIDTH)
+        num_trows = tiff_handle.GetField(T.TIFFTAG_TILELENGTH)
+
+        if num_tcols and num_trows:
```

Comment:
Don't check for the tile_shape here, but with the computation of maxzoom (or move both here). Also, either both maxzoom and tile_shape are set, or none. The tile_shape can be passed directly it to DataArrayShadow.

Also, make sure to remove .getTile() in DataArrayShadow, if tile_shape is None.